compio-fs/src/stdio/unix.rs

```diff
@@ -0,0 +1,111 @@
+use std::{io, mem::ManuallyDrop};
+
+use compio_buf::{BufResult, IoBuf, IoBufMut};
+use compio_driver::{AsRawFd, FromRawFd, RawFd};
+use compio_io::{AsyncRead, AsyncWrite};
+use compio_runtime::TryAsRawFd;
+
+use crate::pipe::{Receiver, Sender};
+
+/// A handle to the standard input stream of a process.
+///
+/// See [`stdin`].
+pub struct Stdin(ManuallyDrop<Receiver>);
+
+impl Stdin {
+    pub(crate) fn new() -> Self {
+        Self(ManuallyDrop::new(unsafe {
+            Receiver::from_raw_fd(io::stdin().as_raw_fd())
```

Comment:
Whhy not use libc const? This will lock and unlock instantly.

Reply:
Good question. Will the three stdio fd change? Or they are constant even if the user closes some of them?

Reply:
At least on unix it's `libc::STD{IN,OUT,ERR}_FILENO`, and that's what std uses.